llvm/test/tools/sycl-post-link/spec-constants/SYCL-2020-zeroinitializer.ll

```diff
@@ -0,0 +1,104 @@
+%struct._ZTS9composite.composite = type { %struct._ZTS6nested.nested, i8, i8, i64 }
+%struct._ZTS6nested.nested = type { float }
+%"class._ZTSN2cl4sycl5rangeILi1EEE.cl::sycl::range" = type { %"class._ZTSN2cl4sycl6detail5arrayILi1EEE.cl::sycl::detail::array" }
+%"class._ZTSN2cl4sycl6detail5arrayILi1EEE.cl::sycl::detail::array" = type { [1 x i64] }
+%"class._ZTSN2cl4sycl2idILi1EEE.cl::sycl::id" = type { %"class._ZTSN2cl4sycl6detail5arrayILi1EEE.cl::sycl::detail::array" }
+
+$"_ZTSZZ4mainENK3$_0clERN2cl4sycl7handlerEE7Kernel1" = comdat any
+
+@__builtin_unique_stable_name._ZN2cl4sycl14kernel_handler33getSpecializationConstantOnDeviceIL_ZL6int_idEiLPv0EEET0_v = private unnamed_addr addrspace(1) constant [70 x i8] c"_ZTSN2cl4sycl6detail32specialization_id_name_generatorIL_ZL6int_idEEE\00", align 1
+@_ZL6int_id = internal addrspace(1) constant %"class._ZTSN2cl4sycl17specialization_idIiEE.cl::sycl::specialization_id" zeroinitializer, align 4
+@__builtin_unique_stable_name._ZN2cl4sycl14kernel_handler33getSpecializationConstantOnDeviceIL_ZL9double_idEdLPv0EEET0_v = private unnamed_addr addrspace(1) constant [73 x i8] c"_ZTSN2cl4sycl6detail32specialization_id_name_generatorIL_ZL9double_idEEE\00", align 1
+@_ZL9double_id = internal addrspace(1) constant %"class._ZTSN2cl4sycl17specialization_idIdEE.cl::sycl::specialization_id" zeroinitializer, align 8
+@__builtin_unique_stable_name._ZN2cl4sycl14kernel_handler33getSpecializationConstantOnDeviceIL_ZL12composite_idE9compositeLPv0EEET0_v = private unnamed_addr addrspace(1) constant [77 x i8] c"_ZTSN2cl4sycl6detail32specialization_id_name_generatorIL_ZL12composite_idEEE\00", align 1
+@_ZL12composite_id = internal addrspace(1) constant %"class._ZTSN2cl4sycl17specialization_idI9compositeEE.cl::sycl::specialization_id" zeroinitializer, align 8
+
+; Function Attrs: convergent norecurse
+define weak_odr dso_local spir_kernel void @"_ZTSZZ4mainENK3$_0clERN2cl4sycl7handlerEE7Kernel1"(i32 addrspace(1)* %_arg_, %"class._ZTSN2cl4sycl5rangeILi1EEE.cl::sycl::range"* byval(%"class._ZTSN2cl4sycl5rangeILi1EEE.cl::sycl::range") align 8 %_arg_1, %"class._ZTSN2cl4sycl5rangeILi1EEE.cl::sycl::range"* byval(%"class._ZTSN2cl4sycl5rangeILi1EEE.cl::sycl::range") align 8 %_arg_2, %"class._ZTSN2cl4sycl2idILi1EEE.cl::sycl::id"* byval(%"class._ZTSN2cl4sycl2idILi1EEE.cl::sycl::id") align 8 %_arg_3) local_unnamed_addr #0 comdat !kernel_arg_buffer_location !4 {
+entry:
+  %ref.tmp.i = alloca %struct._ZTS9composite.composite, align 8
+  %0 = getelementptr inbounds %"class._ZTSN2cl4sycl2idILi1EEE.cl::sycl::id", %"class._ZTSN2cl4sycl2idILi1EEE.cl::sycl::id"* %_arg_3, i64 0, i32 0, i32 0, i64 0
+  %1 = addrspacecast i64* %0 to i64 addrspace(4)*
+  %2 = load i64, i64 addrspace(4)* %1, align 8
+  %add.ptr.i = getelementptr inbounds i32, i32 addrspace(1)* %_arg_, i64 %2
+  %ref.tmp.ascast.i = addrspacecast %struct._ZTS9composite.composite* %ref.tmp.i to %struct._ZTS9composite.composite addrspace(4)*
```

Comment:
since this is not E2E test anyway, please simplify it and leave only (mostly) the IR which is supposed to be changed (and the change verified).

Reply:
Simplified the test in 1a9927894d4ad79151f89ff742454c848849af68